src/Tasks/Microsoft.NET.Build.Tasks/GenerateDepsFile.cs

```diff
@@ -111,5 +126,115 @@ protected override void ExecuteCore()
             _filesWritten.Add(new TaskItem(DepsFilePath));
 
         }
+
+        private void LoadFilesToSkip()
+        {
+            foreach (var fileToSkip in FilesToSkip)
+            {
+                string packageId, packageSubPath;
+                ConflictResolution.NuGetUtilities.GetPackageParts(fileToSkip.ItemSpec, out packageId, out packageSubPath);
+
+                if (String.IsNullOrEmpty(packageId) || String.IsNullOrEmpty(packageSubPath))
+                {
+                    continue;
+                }
+
+                var itemType = fileToSkip.GetMetadata(nameof(ConflictResolution.ConflictItemType));
+                var packagesWithFilesToSkip = (itemType == nameof(ConflictResolution.ConflictItemType.Reference)) ? compileFilesToSkip : runtimeFilesToSkip;
+
+                HashSet<string> filesToSkipForPackage;
+                if (!packagesWithFilesToSkip.TryGetValue(packageId, out filesToSkipForPackage))
+                {
+                    packagesWithFilesToSkip[packageId] = filesToSkipForPackage = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
+                }
+
+                filesToSkipForPackage.Add(packageSubPath);
+            }
+        }
+
+        private DependencyContext TrimFilesToSkip(DependencyContext sourceDeps)
+        {
+            return new DependencyContext(sourceDeps.Target,
+                                         sourceDeps.CompilationOptions,
+                                         TrimCompilationLibraries(sourceDeps.CompileLibraries),
+                                         TrimRuntimeLibraries(sourceDeps.RuntimeLibraries),
+                                         sourceDeps.RuntimeGraph);
+        }
+
+        private IEnumerable<RuntimeLibrary> TrimRuntimeLibraries(IReadOnlyList<RuntimeLibrary> runtimeLibraries)
+        {
+            foreach (var runtimeLibrary in runtimeLibraries)
+            {
+                HashSet<string> filesToSkip;
+                if (runtimeFilesToSkip.TryGetValue(runtimeLibrary.Name, out filesToSkip))
+                {
+                    yield return new RuntimeLibrary(runtimeLibrary.Type,
```

Comment:
I can go either way.

But the reason I brought it up is because if NuGet was doing this conflict resolution, you wouldn't see this library at all in the `project.assets.json` file, thus it wouldn't be in the deps file.  However, since NuGet can't do this resolution (because we are collapsing the System.XXX libraries into Microsoft.NETCore.App), these old System.XXX v4 libraries are showing up in the deps file.